### DISCUSSION TITLE
Export as XML

### DISCUSSION THREAD

**Discussion by sunnyTaner** [maxGraph/maxGraph#523] [Q&A] [answered]
Hi!I am trying to export as XML,but there was a problem. How can I do it
 ```
const encoder = new Codec();
 const result = encoder.encode(graph.getDataModel());
xmlUtils.getPrettyXml(result)
```
This doesn't work and code error reported. The error is as follows
![image](https://github.com/user-attachments/assets/45096f22-da37-4b53-a38d-56cc7748e7c4)

**Comment by tbouffard** [comment, marked as answer]
Hello, the error occurs when no codec is registered.

From [version 0.6.0](https://github.com/maxGraph/maxGraph/releases/tag/v0.6.0) of `maxGraph`, codecs supplied by maxGraph are no longer registered by default, they **MUST** be registered before performing an `encode` or `decode`

For example:
- You can use the `registerModelCodecs` function (or other related functions) to register the Model codecs.
- To serialize the `maxGraph` model, you can use the `ModelXmlSerializer` class, which registers codecs under the hood and provide a simpler syntax.

Is that clear?

In addition, I'm working on adding a page dedicated to this topic in documentation #524.

**Reply by sunnyTaner** [reply]
According to your answer, it has been resolved.This is very helpful!Thanks a lot！非常感谢！

**Reply by tbouffard** [reply]
Perfect, I have marked this question as answered and I will update the documentation as mentioned earlier.
